### ISSUE TITLE
Can't load BERTopic model

### ISSUE THREAD

**Issue by anubhabdaserrr** [MaartenGr/BERTopic#883]
Ever since Google Colab upgraded from Python 3.7 to 3.8, I can't seem to load my saved BERTopic model which trained on Google Colab prior to the upgrade.

<img width="663" alt="Screenshot 2022-12-18 at 12 37 25 PM" src="https://user-images.githubusercontent.com/55646192/208285924-7e4cc949-dfdc-4094-8492-5a12f42c9069.png">

Also, the "use fallback runtime version" feature is not available in the Command Palette anymore [It was available only up until Mid-December].

Is there, any possible way to load the old BERTopic model in Google Colab? I can't figure out the exact source of conflict.

<img width="726" alt="Screenshot 2022-12-18 at 12 32 35 PM" src="https://user-images.githubusercontent.com/55646192/208285772-9a96d5a6-738d-4f47-8b3e-f92a7ce80027.png">

I do have the requirements.txt file from the old run and tried updating joblib to the original version, but I still can't seem to get it to work.

**Comment by ubayram** [comment]
Having exactly the same error on Colab when loading a saved model. There were no problems with saving the model on the same environment, but just loading it is throwing the TypeError as above. For a temporary solution, will try downgrading the python version of Colab to 3.7 for now.

**Comment by sarisemih** [comment]
I get the same error after running it on **python 3.7**
I ran these codes:
`!sudo apt-get update -y`
`!sudo apt-get install python3.7`
`!sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.7 1`
`!sudo update-alternatives --config python3`
`!apt-get install python3-pip`
`!python -m pip install --upgrade pip --user`
`!pip install bertopic`

**Comment by sarisemih** [comment]
I ran these codes to try in Python 3.7 version. I'm having trouble importing BERTopic.

!sudo apt-get update -y
!sudo apt-get install python3.7
!sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.7 1
!sudo update-alternatives --config python3
!apt-get install python3-pip
!python -m pip install --upgrade pip --user
!pip install bertopic

**ERROR**: Could not build wheels for hdbscan, which is required to install pyproject.toml-based projects.

**Comment by anubhabdaserrr** [comment]
I faced the same problem as well. You could try running it on Kaggle since Kaggle uses Python 3.7. However, Kaggle (by default) uses an older version of joblib library (Which seems to be the origin of the conflict from the above traceback). If you have the requirements.txt file from the Google Colab training run, you could try finding joblib version and upgrading the package on kaggle as follows.

`!pip install joblib==1.2.0`

Basically, upload the saved model & your dataset on Kaggle, install BERTopic & upgrade joblib; restart the runtime, and it should probably work, as my model was loading without any errors, however I still couldn't get it to work, since my BERTopic saved model is above 11 GB. If you have a smaller model, you could try this method to see if it works.

**Comment by RanAl2020** [comment]
how can I get the requirements.txt file from the Google Colab?

**Comment by anubhabdaserrr** [comment]
Run the following command in a cell

`! pip freeze > requirements.txt`

You needed to have the requirements.txt file from your BERTopic training run. Right now, it wouldn't work because Colab has upgraded Python version. Try updating joblib to version 1.2.0 to see if it works.

**Comment by RanAl2020** [comment]
thank you, is there any solution to get back our model using Colab?

**Comment by MaartenGr** [comment]
Due to inactivity, I'll be closing this issue. Let me know if you want me to re-open the issue!
